src/topp/FileMerger.cpp

```diff
@@ -245,8 +245,18 @@ class TOPPFileMerger :
     {
       ConsensusMap out;
       ConsensusXMLFile fh;
+      // load the metadata from the first file
       fh.load(file_list[0], out);
-      // skip first file
+      // but annotate the origins
+      if (annotate_file_origin)
+      {
+        for (ConsensusMap::iterator it = out.begin(); it != out.end(); ++it)
```

Comment:
We could think about using for_each and a lambda in the future.
in my view its even easier to read